lib/foreman_maintain/cli/upgrade_command.rb

```diff
@@ -1,85 +1,72 @@
 module ForemanMaintain
   module Cli
     class UpgradeCommand < Base
-      def self.target_version_option
-        option '--target-version', 'TARGET_VERSION', 'Target version of the upgrade',
-          :required => false
-      end
-
       def self.disable_self_upgrade_option
         option '--disable-self-upgrade', :flag, 'Disable automatic self upgrade',
           :default => false
       end
 
-      def current_target_version
-        current_target_version = ForemanMaintain::UpgradeRunner.current_target_version
-        if current_target_version && target_version && target_version != current_target_version
-          raise Error::UsageError,
-            "Can't set target version #{target_version}, "\
-            "#{current_target_version} already in progress"
-        end
-        @target_version = current_target_version if current_target_version
-        return true if current_target_version
-      end
-
-      def validate_target_version!
-        return if current_target_version
-        unless UpgradeRunner.available_targets.include?(target_version)
-          message_start = if target_version
-                            "Can't upgrade to #{target_version}"
-                          else
-                            '--target-version not specified'
-                          end
-          message = <<~MESSAGE
-            #{message_start}
-            Possible target versions are:
-          MESSAGE
-          versions = UpgradeRunner.available_targets.join("\n")
-          raise Error::UsageError, message + versions
-        end
-      end
-
       def upgrade_runner
         return @upgrade_runner if defined? @upgrade_runner
-        validate_target_version!
-        @upgrade_runner = ForemanMaintain::UpgradeRunner.new(target_version, reporter,
+        @upgrade_runner = ForemanMaintain::UpgradeRunner.new(reporter,
           :assumeyes => assumeyes?,
           :whitelist => whitelist || [],
           :force => force?).tap(&:load)
       end
 
-      def print_versions(target_versions)
-        target_versions.sort.each { |version| puts version }
-      end
-
       def allow_self_upgrade?
         !disable_self_upgrade?
       end
 
+      def try_upgrade
+        if upgrade_runner.available?
+          yield
+        else
+          instance = ForemanMaintain.detector.feature(:instance)
+          msg = <<~BANNER
+
+            There are no upgrades available.
+            The current version of #{instance.product_name} is #{instance.current_version}.
```

Comment:
Currently, this prints `X.Y.Z` style versions, which is not entirely correct when there is a .z release available (it tells me 3.13.0 is current, while really 3.13.1 is out). Should we only print X.Y here?

My ACK stands regardless of the outcome of this discussion.

Reply:
Discussed offline to implement that and done so to speed things up.